WordPress/src/main/java/org/wordpress/android/ui/accounts/login/WPcomLoginHelper.kt

```diff
@@ -0,0 +1,51 @@
+package org.wordpress.android.ui.accounts.login
+
+import android.net.Uri
+import android.util.Log
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.cancel
+import kotlinx.coroutines.runBlocking
+import org.wordpress.android.BuildConfig
+import org.wordpress.android.fluxc.network.rest.wpapi.WPcomLoginClient
+import org.wordpress.android.fluxc.store.AccountStore
+import org.wordpress.android.util.config.AppConfig
+import javax.inject.Inject
+import kotlin.coroutines.CoroutineContext
+
+class WPcomLoginHelper @Inject constructor(
+    private val loginClient: WPcomLoginClient,
+    private val accountStore: AccountStore,
+    private val appConfig: AppConfig
+) {
+    private val context: CoroutineContext = Dispatchers.IO
+
+    fun loginUri(): Uri {
+        return loginClient.loginUri(BuildConfig.OAUTH_REDIRECT_URI)
+    }
+
+    fun tryLoginWithDataString(data: String?) {
+        if (data == null) {
+            return
+        }
+
+        val code = this.codeFromAuthorizationUri(data) ?: return
+
+        runBlocking {
+            val tokenResult = loginClient.exchangeAuthCodeForToken(code, BuildConfig.OAUTH_REDIRECT_URI)
+            accountStore.updateAccessToken(tokenResult.getOrThrow())
```

Comment:
I think you'd want to check if the token belongs to the current user before saving it. Users can log in with any account from the login webpage.

Reply:
This should be for first-time authentication only – I'm thinking we'll want a `reauthenticate` method that does what you say – forces login to a specific account